### ISSUE TITLE
Not sure why my integration isn't getting any data from Manchester City Council...?

### ISSUE THREAD

**Issue by jonxyz** [mampfes/hacs_waste_collection_schedule#756]
### Discussed in https://github.com/mampfes/hacs_waste_collection_schedule/discussions/755

<div type='discussions-op-text'>

<sup>Originally posted by **jonxyz** March 12, 2023</sup>
Hi there

Love the work you've put into this integration, but i'm having some trouble getting it working and i'd appreciate any help you can offer...! 🙂 

My config is very simple at the moment:

```
# Waste Collection Config
waste_collection_schedule:
    sources:
    - name: manchester_uk
      args:
        uprn: "xxxxxxxxxxx" 
```

I've redacted my 11 digit UPRN above for privacy reasons, but i've tried it with and without a leading 0, the manchester site html code has it with the leading 0 and the online UPRN search has it without the 0.

But in both cases i'm getting the following error in the logs, which implies no data being returned, but i'm not sure what i'm missing or where to go next?

```
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
Integration: waste_collection_schedule (documentation)
First occurred: 17:28:13 (1 occurrences)
Last logged: 17:28:13

fetch failed for source Manchester City Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/manchester_uk.py", line 48, in fetch collection_type = img_tag["alt"] TypeError: 'NoneType' object is not subscriptable
```</div>

**Comment by dt215git** [comment]
Looks like it's failing to find an image and extract its alternative text - when you look at your schedule on the web site, are you seeing icons/images for each waste type?

The script also looks to be limited to capturing the following waste types:
- Black / Grey Bin
- Blue Bin
- Brown Bin
- Green Bin

If your have other waste types, then it might be ignoring them.

These images illustrate the above differences.

![image](https://user-images.githubusercontent.com/86194065/224568640-9a60c979-e4de-4581-91ea-a7e090a78f75.png) ![image](https://user-images.githubusercontent.com/86194065/224568678-52171546-212c-4766-90b5-cc74a2ad2740.png)

**Comment by jonxyz** [comment]
Aha! Thanks for the input that's helpful...to at least understand the issue...even if it may be somewhat beyond me to solve it (I have zero Python skills) 

My output on the website has no images at all, I live in an apartment so my bins are all of the "Large Container type" which as you say it's probably not looking for. Because i'm in an apartment I don't actually have to put the bins out (which is great!) but it's useful to know what the next collections are, as the Large Brown Bin and the Large Blue bin alternate weeks and so there's usually no space left in them towards the end of a cycle.

Here's my output:
<img width="681" alt="image" src="https://user-images.githubusercontent.com/505304/224765467-a6852588-c2af-4c67-8c90-ecf5d3abd3a7.png">

I'll see whether I can figure it out, but if anyone out there fancies helping with the coding it would be much appreciated!

**Comment by jonxyz** [comment]
Your nudge was just what I needed...after some messing around I got it working! Thanks!

In case it's helpful to anyone else, here's my tweaked manchester_uk.py

`import logging
from datetime import datetime

import requests
from bs4 import BeautifulSoup
from waste_collection_schedule import Collection  # type: ignore[attr-defined]

TITLE = "Manchester City Council"
DESCRIPTION = "Source for bin collection services for Manchester City Council, UK."
URL = "https://www.manchester.gov.uk"
TEST_CASES = {
    "domestic": {"uprn": "000077065560"},
}

API_URL = "https://www.manchester.gov.uk/bincollections/"
ICON_MAP = {
    "Black / Grey Bin": "mdi:trash-can",
    "Blue Bin": "mdi:recycle",
    "Brown Bin": "mdi:glass-fragile",
    "Green Bin": "mdi:leaf",
    "Large Blue Container": "mdi:package-variant",
    "Large Brown Container": "mdi:bottle-wine-outline",
    "Large Domestic Waste Container": "mdi:trash-can-outline",
}

_LOGGER = logging.getLogger(__name__)


class Source:
    def __init__(self, uprn: int):
        self._uprn = uprn

    def fetch(self):
        entries = []

        r = requests.post(
            API_URL,
            data={"mcc_bin_dates_uprn": self._uprn, "mcc_bin_dates_submit": "Go"},
        )

        soup = BeautifulSoup(r.text, features="html.parser")
        results = soup.find_all("div", {"class": "collection"})

        for result in results:
            date = result.find("p", {"class": "caption"})
            dates = []
            dates.append(str(date.text).replace("Next collection ", "", 1))
            for date in result.find_all("li"):
                dates.append(date.text)
            h3_tag = result.find("h3").text.replace("  DUE TODAY", "").strip()
            collection_type = h3_tag
            for current_date in dates:
                date = datetime.strptime(current_date, "%A %d %b %Y").date()
                entries.append(
                    Collection(
                        date=date,
                        t=collection_type,
                        icon=ICON_MAP.get(collection_type),
                    )
                )

        return entries
`

**Comment by dt215git** [comment]
Have made some changes: #758

**Comment by jonxyz** [comment]
Superstar, that's great, thanks so much!

**Comment by mampfes** [comment]
fixed by #758
